lib/Parse/DNS/Zone.pm

```diff
@@ -408,6 +413,42 @@ sub _load_zonefile {
 	return do { local $/; <$zonefh> }; # slurp
 }
 
+sub _parse_time_unit {
+	my $string = shift;
+
+	my $time_units_regex = qr/^
+			(?: (?=\d+(?: w|d|h|m|s))
+			(\d+w)? (\d+d)? (\d+h)? (\d+m)? (\d+s)? ) | (\d+)
+		$/xi;
+	my @parts = ($string =~ $time_units_regex);
+
+	my %multi_for_unit = (
+		s => 1,
+		m => 60,
+		h => 3600,
+		d => 86400,
+		w => 604800,
+	);
+
+	my $seconds = 0;
+
+	for my $part (@parts) {
+		next
+			unless $part;
+		my ($value, $unit) = $part =~ /^(\d+)(\D+)?$/;
+		$unit = 's'
+			unless $unit;
+		# from Bind reference manual:
+		# Both TTL-style and ISO 8601 duration formats are case-insensitive.
+		$unit = lc($unit);
+		die "unit $unit not supported"
```

Comment:
Perhaps change the error message to `BUG: Known unit is unhandled`, or what do you think? Makes it clear that it isn't supposed to happen.

(But I'm still pondering if it is a good idea to include it - this can never happen, as the regexp already validates the supported units. Only way for this to happen if the regexp and the hash diverges. But I can't imagine this set will ever change? And if so, having unit tests to cover it should be enough. But I won't force the issue.)